flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/FailoverStrategy.java

```diff
@@ -34,4 +34,22 @@
 	 * @return set of IDs of vertices to restart
 	 */
 	Set<ExecutionVertexID> getTasksNeedingRestart(ExecutionVertexID executionVertexId, Throwable cause);
+
+	// ------------------------------------------------------------------------
+	//  factory
+	// ------------------------------------------------------------------------
```

Comment:
position marker is not a good choice (see Clean Code by Robert C Martin, page 67, Chapter 4).

Reply:
Sorry but I don't have that book.
Let's just keep it as this marker usage is widely used in Flink.